core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -461,7 +461,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
           appListener.appAttemptId,
           appListener.startTime.getOrElse(-1L),
           appListener.endTime.getOrElse(-1L),
-          fileStatus.getModificationTime(),
+          clock.getTimeMillis(),
```

Comment:
Hi @vanzin, we can achieve this by simply doing `if (appCompleted) fileStatus.getModificationTime else clock.getTimeMillis()`.

But I am worried about this scenario: If we assume that modtime will not change when job is completed, the last updated time will move forward while job is in progress and it will go back when job is completed.

But then I found [this code](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala#L230) which is executed when we complete the job. We call `fileSystem.setTimes` after `fileSystem.rename`. 

So seems like my assumption is not a valid case. Let me think about it more/add more tests etc..